boards/posix/nrf52_bsim/Kconfig.defconfig

```diff
@@ -0,0 +1,62 @@
+
+if BOARD_NRF52_BSIM
+
+config BUILD_OUTPUT_BIN
+	default n
+
+config BUILD_OUTPUT_EXE
+	default y
+
+config BOARD
+	default "nrf52_bsim"
+
+#quite ugly hack: we need these symbols below to be able to compile the HAL,
+#but we would not get them otherwise as we do not really inherit this SOC
+#configuration (and we cannot pull the ARM related configs)
+
+#config SOC_FAMILY_NRF
```

Comment:
Note that these commented configs are here only so it is evident what was needed to make it work without all the #if changes. Having these re-definitions was ok while the board was off-tree. But when bringing the board in, the changes in the "kconfig fix" commit are necessary.
This commit would be squashed into the board one, and this comments would be removed before merging.